apps/08jquery-towers-of-hanoi/script.js

```diff
@@ -1,5 +1,38 @@
 'use strict';
-
+//made $block global and =null for following if/else conditions to pull from
+var $block = null;
 $(document).ready(function() {
-  // Put app logic here
+
+  $('[data-stack]').on('click', function() {
+    // setting #announce-game-win to empty string at every click to clear out any denial messages from previous click attempt.
+    $('#announce-game-won').text('')
+    if ($block === null) {
+      if ($(this).children().length < 1) {
+        //give alert if user is trying to pick up from an empty stack
+        $('#announce-game-won').text('Cannot pick up a block from an empty stack.');
+      } else {
+        $block = $(this).children().last().detach()
+      }
+    } else {
+      //allow stack to be appended only if stack is empty or last block of stack is larger than block 'in hand'
+      if ($($block).data('block') < $(this).children().last().data('block') || $(this).children().length < 1) {
+        $(this).append($block);
+        $block = null;
```

Comment:
If I'm being **super** picky, I might say a comment here would help explain why we're setting this to `null`